examples-gallery/advanced/plot_particle_in_tube.py

```diff
@@ -463,66 +514,82 @@ def plot_3d_circle(ax, center, radius, normal, num_points=100):
     ax.plot(circle_3d[0, :], circle_3d[1, :], circle_3d[2, :], color='blue')
 
 
+class Vector3D(FancyArrowPatch):
+    """Vector that can be animated in 3D."""
+
+    def __init__(self, xyz_tail, xyz_vec, *args, **kwargs):
+        super().__init__((0, 0), (0, 0), *args, **kwargs)
+        self._xyz_tail = xyz_tail
+        self._xyz_vec = xyz_vec
+
+    def do_3d_projection(self, renderer=None):
+        xs, ys, zs = proj_transform(
+            (self._xyz_tail[0], self._xyz_tail[0] + self._xyz_vec[0]),
+            (self._xyz_tail[1], self._xyz_tail[1] + self._xyz_vec[1]),
+            (self._xyz_tail[2], self._xyz_tail[2] + self._xyz_vec[2]),
+            self.axes.M)
+        self.set_positions((xs[0], ys[0]), (xs[1], ys[1]))
+        return min(zs)
+
+    def set_data(self, xyz_tail, xyz_vec):
+        self._xyz_tail = xyz_tail
+        self._xyz_vec = xyz_vec
```

Comment:
I changed to this given that quiver locations cannot be animated. But now the black dot and black curve are not showing.